crates/ruff_server/src/session.rs

```diff
@@ -147,24 +149,32 @@ impl Session {
         Ok(())
     }
 
-    pub(crate) fn num_documents(&self) -> usize {
-        self.index.num_documents()
+    pub(crate) fn resolved_client_capabilities(&self) -> &ResolvedClientCapabilities {
+        &self.resolved_client_capabilities
     }
 
-    pub(crate) fn num_workspaces(&self) -> usize {
-        self.index.num_workspaces()
+    pub(crate) fn encoding(&self) -> PositionEncoding {
+        self.position_encoding
     }
 
-    pub(crate) fn list_config_files(&self) -> Vec<&std::path::Path> {
-        self.index.list_config_files()
+    /// Returns an iterator over the paths to the configuration files in the index.
+    pub(crate) fn config_file_paths(&self) -> impl Iterator<Item = &Path> {
+        self.index.config_file_paths()
     }
 
-    pub(crate) fn resolved_client_capabilities(&self) -> &ResolvedClientCapabilities {
-        &self.resolved_client_capabilities
+    /// Returns the resolved global client settings.
+    pub(crate) fn global_client_settings(&self) -> ResolvedClientSettings {
+        ResolvedClientSettings::global(&self.global_settings)
     }
 
-    pub(crate) fn encoding(&self) -> PositionEncoding {
-        self.position_encoding
+    /// Returns the number of open documents in the session.
+    pub(crate) fn open_documents(&self) -> usize {
```

Comment:
Nit: I think I prefer `open_documents_len` to make it explicit that this doesn't return the open documents itself.